kotlin/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.kt

```diff
@@ -0,0 +1,355 @@
+
+import android.graphics.Color
+import android.os.Bundle
+import android.util.Log
+import android.view.MotionEvent
+import android.widget.SeekBar
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.geoanalysis.LocationViewshed
+import com.esri.arcgisruntime.geoanalysis.Viewshed
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.layers.ArcGISSceneLayer
+import com.esri.arcgisruntime.mapping.ArcGISScene
+import com.esri.arcgisruntime.mapping.ArcGISTiledElevationSource
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Surface
+import com.esri.arcgisruntime.mapping.view.AnalysisOverlay
+import com.esri.arcgisruntime.mapping.view.Camera
+import com.esri.arcgisruntime.mapping.view.DefaultSceneViewOnTouchListener
+import com.esri.arcgisruntime.mapping.view.OrbitLocationCameraController
+import java.util.concurrent.ExecutionException
+import kotlin.math.roundToInt
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.viewshed_seek_bars.*
+
+
+class MainActivity : AppCompatActivity() {
+
+  companion object {
+    private val TAG: String = MainActivity::class.java.simpleName
+  }
+
+  private val initHeading = 0
+  private val initPitch = 60
+  private val initHorizontalAngle = 75
+  private val initVerticalAngle = 90
+  private val initMinDistance = 0
+  private val initMaxDistance = 1500
+
+  private var minDistance: Int = 0
+  private var maxDistance: Int = 0
+
+  private lateinit var viewShed: LocationViewshed
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a surface for elevation data
+    val surface = Surface().apply {
+      elevationSources.add(ArcGISTiledElevationSource(getString(R.string.elevation_service)))
+    }
+
+    // create a layer of buildings
+    val buildingsLayer = ArcGISSceneLayer(getString(R.string.buildings_layer))
+
+    // create a scene and add imagery basemap, elevation surface, and buildings layer to it
+    val buildingsScene = ArcGISScene().apply {
+      basemap = Basemap.createImagery()
+      baseSurface = surface
+      operationalLayers.add(buildingsLayer)
+    }
+
+    val initLocation = Point(-4.50, 48.4, 1000.0)
+    Viewshed.setFrustumOutlineColor(Color.BLUE)
+
+    // create viewshed from the initial location
+    viewShed = LocationViewshed(
+      initLocation,
+      initHeading.toDouble(),
+      initPitch.toDouble(),
+      initHorizontalAngle.toDouble(),
+      initVerticalAngle.toDouble(),
+      initMinDistance.toDouble(),
+      initMaxDistance.toDouble()
+    ).apply {
+      setFrustumOutlineVisible(true)
+    }
+
+    sceneView.apply {
+      //add the buildings scene to the sceneView
+      scene = buildingsScene
+      // add a camera and set it to orbit the location point of the frustum
```

Comment:
I noticed that if you double tap and drag the frustum away, the camera doesn't orbit the location point of the frustum. therefore we might want to alter this comment slightly to 
```suggestion
      // add a camera and set it to orbit the starting location point of the frustum
```